packages/@tailwindcss-upgrade/src/utils/packages.ts

```diff
@@ -0,0 +1,61 @@
+import { execSync } from 'node:child_process'
+import fs from 'node:fs/promises'
+import { dirname, resolve } from 'node:path'
+import { warn } from './renderer'
+
+let didWarnAboutPackageManager = false
+
+export async function pkg(command: string, base: string): Promise<Buffer | void> {
+  let packageManager = await detectPackageManager(base)
+  if (!packageManager) {
+    if (!didWarnAboutPackageManager) {
+      didWarnAboutPackageManager = true
+      warn('Could not detect a package manager. Please manually update `tailwindcss` to v4.')
+    }
+    return
+  }
+  return execSync(`${packageManager} ${command}`, {
+    cwd: base,
+  })
+}
+
+async function detectPackageManager(base: string): Promise<null | string> {
+  do {
+    // 1. Check package.json for a `packageManager` field
+    let packageJsonPath = resolve(base, 'package.json')
+    try {
+      let packageJsonContent = await fs.readFile(packageJsonPath, 'utf-8')
+      let packageJson = JSON.parse(packageJsonContent)
+      if (packageJson.packageManager) {
+        if (packageJson.packageManager.includes('yarn')) {
+          return 'yarn'
+        }
+        if (packageJson.packageManager.includes('pnpm')) {
+          return 'pnpm'
+        }
+        if (packageJson.packageManager.includes('npm')) {
+          return 'npm'
+        }
+      }
+    } catch {}
+
+    // 2. Check for common lockfiles
```

Comment:
We can probably do this in a cleaner way. But bun is switching from `bun.lockb` (binary) to a `bun.lock` (text) file.
```suggestion
    // 2. Check for common lockfiles
    try {
      await fs.access(resolve(base, 'bun.lockb'))
      return 'bun'
    } catch {}
    try {
      await fs.access(resolve(base, 'bun.lock'))
      return 'bun'
    } catch {}
```